samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/queue/delivery/TimedAssessmentRunnable.java

```diff
@@ -102,8 +103,22 @@ public void run(){
           Integer extendedTime = null;
 
           // The specific student has more time than the thread knows about
-          if (assessmentExtended != null && assessmentExtended.hasExtendedTime()) {
+          if (assessmentExtended.hasExtendedTime()) {
             extendedTime = assessmentExtended.getTimeLimit();
+            
+            // Check if this is a late exception (starting after due date but before retract date)
+            Date assessmentDueDate = publishedAssessment.getDueDate();
+            Date extendedDueDate = assessmentExtended.getDueDate();
+            Date attemptDate = ag.getAttemptDate();
+            // If the student has an extended due date, and they started after the regular due date
+            if (extendedDueDate != null && assessmentDueDate != null && 
+                attemptDate != null && extendedDueDate.after(assessmentDueDate) && 
+                attemptDate.after(assessmentDueDate) && attemptDate.before(extendedDueDate)) {
+              // Allow student to take the assessment with their extended time
+              log.info("SAMIGO_TIMED_ASSESSMENT:LATE_EXCEPTION ID:{} user_id:{} attempt_after_due_date", this.timedAGId, ag.getAgentId());
```

Comment:
is info level to noisy?

Reply:
kind of yeah, but it's where the other log messages are